proto/data.proto

```diff
@@ -133,6 +133,12 @@ message ChangeReplicasTrigger {
   repeated Replica updated_replicas = 4 [(gogoproto.nullable) = false];
 }
 
+// CommitTrigger encapsulates all of the internal-only commit triggers.
+message InternalCommitTrigger {
+  optional SplitTrigger split_trigger = 1;
```

Comment:
Gogoproto has `onlyone`, which is similar, but I don't think it really helps here. We'd just have to do a type switch on the result anyway.